lib/ansible/modules/windows/win_dns_client.ps1

```diff
@@ -193,29 +209,29 @@ Try {
 
     Write-DebugLog ("Validating adapter name {0}" -f $adapter_names)
 
-    $adapters = @($adapter_names)
-
-    If($adapter_names -eq "*") {
-        $adapters = Get-NetAdapter | Select-Object -ExpandProperty Name
-    }
     # TODO: add support for an actual list of adapter names
     # validate network adapter names
-    ElseIf(@(Get-NetAdapter | Where-Object Name -eq $adapter_names).Count -eq 0) {
-        throw "Invalid network adapter name: {0}" -f $adapter_names
+    $adapter_names = @($adapter_names)
+
+    foreach($adapter_name in $adapter_names) {
+        If($adapter_name -eq "*") {
+            $adapters = Get-NetAdapter | Select-Object -ExpandProperty Name
+        } ElseIf(@(Get-NetAdapter | Where-Object {$_.Name -like $adapter_names}).Count -eq 0) {
+            throw "Invalid network adapter name: {0}" -f $adapter_name
+        }
     }
 
-    Write-DebugLog ("Validating IP addresses ({0})" -f ($ipv4_addresses -join ", "))
+    Write-DebugLog ("Validating IP addresses ({0})" -f ($dns_servers -join ", "))
 
-    $invalid_addresses = @($ipv4_addresses | ? { -not (Validate-IPAddress $_) })
+    $invalid_addresses = @($dns_servers | Where-Object { -not (Assert-IPAddress $_) })
 
-    If($invalid_addresses.Count -gt 0) {
+    If([bool]$invalid_addresses) {
         throw "Invalid IP address(es): ({0})" -f ($invalid_addresses -join ", ")
```

Comment:
Maybe this should be a warning since you check it in other places and never use invalid ips?

Reply:
Are you sure, shouldn't the play fail if an invalid address is specified?
The checks on other places are just to avoid weird behaviours, even though they should never be hit.
This one is part of ansible play error handling.

Reply:
If there is a list of IPs and some are bad I would want to get a warning and set the good ones as a user

Reply:
One could (and I think also should) use the ipaddress jinja2 filters to validate and if you hardcoded values at least I want it to stick to a fail fast and fail forward approach...